runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -126,18 +134,26 @@ public class MqttKafkaSubscribeFactory implements MqttKafkaStreamFactory
     private final MutableDirectBuffer subscriptionIdsBuffer;
     private final MutableDirectBuffer filterBuffer;
     private final BindingHandler streamFactory;
+    private final Signaler signaler;
     private final BufferPool bufferPool;
     private final LongUnaryOperator supplyInitialId;
     private final LongUnaryOperator supplyReplyId;
+    private final Supplier<Long> supplyTraceId;
     private final int mqttTypeId;
     private final int kafkaTypeId;
     private final LongFunction<MqttKafkaBindingConfig> supplyBinding;
     private final MqttKafkaHeaderHelper helper;
+    private final int reconnectDelay;
+    private final boolean bootstrapAvailable;
+    private final List<KafkaMessagesBootstrap> bootstrapStreams;
+
+    private int reconnectAttempt;
 
     public MqttKafkaSubscribeFactory(
         MqttKafkaConfiguration config,
         EngineContext context,
-        LongFunction<MqttKafkaBindingConfig> supplyBinding)
+        LongFunction<MqttKafkaBindingConfig> supplyBinding,
+        IntPropertyDef reconnectDelay)
```

Comment:
This constructor parameter would be removed when feedback above has been applied.